src/Ledger/Introduction.lagda

```diff
@@ -60,6 +60,141 @@ Conway~\cite{cip1694} & Complete & Partial & Partial \\
+and the rules for processing them.
+
+We will model this via a number of \textit{state transition systems} (STS) which
+from now on we refer to as ``transition rules'' or just ``rules.''
+These rules describe the different behaviors that determine how the whole system
+evolves and, taken together, they comprise a full description of the ledger protocol.
+Each transition rule consists of the following components:
+\begin{itemize}
+  \item an \textit{environment} consisting of data, read from the ledger state
+        or the outside world, which should be considered constant for the
+        purposes of the rule;
+  \item an \textit{initial state}, consisting of the subset of the full ledger
+        state that is relevant to the rule and which the rule can update;
+  \item a \textit{signal} or \textit{event}, with associated data, that the
+        rule can receive or observe;
+  \item a set of \textit{preconditions} that must be met in order for the transition
+        to be valid;
+  \item a new state that results from the transition rule.
+\end{itemize}
+For example, the UTXOW transition rule defined in \cref{fig:rules:utxow} of
+\cref{sec:witnessing} checks that, among other things, a given transaction is signed
+by the appropriate parties.
+
+The transition rules can be composed in the sense that they may require other
+transition rules to hold as part of their preconditions.  For example, the UTXOW rule
+mentioned above requires the UTXO rule, which checks that the inputs to the
+transaction exist, that the transaction is balanced, and several other conditions.
+
+\begin{figure}[h!]
+  \centering
+  \input{Diagrams/CardanoLedger}
+  \caption{State transition rules of the ledger specification, presented as a
+  directed graph; each node represents a transition rule; an arrow from rule A to
+  rule B indicates that B appears among the premises of A; a dotted arrow represents
+  a dependency in the sense that the output of the target node is an input to the
+  source node, either as part of the source state, the environment or the event
+    (\legendbox{\ConwayColor}~rules added in Conway;
+     \legendbox{\BabbageColor}~rules modified in Conway; dotted ellipses represent rules
+  that are not yet formalized in Agda)
+  }
+  \label{fig:latest-sts-diagram}
+\end{figure}
+
+\begin{figure}[h!]
+{\footnotesize
+\begin{prooftree}
+  \AxiomC{\applyRUpd}
+  \AxiomC{\acceptConds}
+  \AxiomC{ENACT}
+  \BinaryInfC{RATIFY...RATIFY}
+  \UnaryInfC{RATIFIES}
+  \AxiomC{SNAP}
+  \BinaryInfC{EPOCH}
+  \BinaryInfC{NEWEPOCH}
+  \AxiomC{UTXOS}
+  \UnaryInfC{UTXO}
+  \UnaryInfC{UTXOW}
+  \AxiomC{GOV...GOV}
+  \UnaryInfC{GOVS}
+  \AxiomC{DELEG}
+  \AxiomC{GOVCERT}
+  \AxiomC{POOL}
+  \TrinaryInfC{CERT...CERT}
+  \UnaryInfC{CERTS}
+  \TrinaryInfC{LEDGER...LEDGER}
+  \UnaryInfC{LEDGERS}
+  \BinaryInfC{CHAIN}
+\end{prooftree}
+}
+\caption{State transition rules of the ledger specification, presented as a deduction tree.}
+\label{fig:new-chain-diagram}
+\end{figure}
```

Comment:
- Too big to fit in the PDF
- It seems to contain different nodes than the figure above
- Not clear what `LEDGER..LEDGER` means

Reply:
I think I should just delete the deduction tree diagram.